From fb8aaed3b040e589cd880fd714dda5ec00687217 Mon Sep 17 00:00:00 2001
From: Jens Rehsack <sno@netbsd.org>
Date: Mon, 24 Feb 2020 12:10:06 +0100
Subject: [PATCH 2/2] modules: fcntl: allow being detected by importing
 projects

GNU project `make` relies on gnulib but provides some own compatibility
functions - including an `fcntl`, which fails on mingw.
The intension of gnulib is providing these functions and being wider tested,
but silently injecting a function opens battle of compatibility layers.

So adding a hint into target `config.h` to allow deciding whether using
an own compatibility implementation or not.

Signed-off-by: Jens Rehsack <sno@netbsd.org>
---
Upstream-Status: Pending

 m4/gnulib-comp.m4 | 1 +
 1 file changed, 1 insertion(+)

diff --git a/m4/gnulib-comp.m4 b/m4/gnulib-comp.m4
index 3ee0811..cf75541 100644
--- a/m4/gnulib-comp.m4
+++ b/m4/gnulib-comp.m4
@@ -147,6 +147,7 @@
   gl_FUNC_FCNTL
   if test $HAVE_FCNTL = 0 || test $REPLACE_FCNTL = 1; then
     AC_LIBOBJ([fcntl])
+    AC_DEFINE(HAVE_GNULIB_FCNTL, 1, [Define to 1 if you have the `fcntl' function via gnulib.])
   fi
   gl_FCNTL_MODULE_INDICATOR([fcntl])
   gl_FCNTL_H
